tasks/package.py

```diff
@@ -1,31 +1,56 @@
-import glob
 import os
+from datetime import datetime
 
 from invoke import task
 from invoke.exceptions import Exit
 
 from tasks.libs.common.color import color_message
-from tasks.libs.package.size import compute_package_size_metrics
+from tasks.libs.common.git import get_common_ancestor, get_current_branch, get_default_branch
+from tasks.libs.package.size import (
+    PACKAGE_SIZE_TEMPLATE,
+    _get_deb_uncompressed_size,
+    _get_rpm_uncompressed_size,
+    compare,
+    compute_package_size_metrics,
+)
+from tasks.libs.package.utils import (
+    display_message,
+    get_package_path,
+    list_packages,
+    retrieve_package_sizes,
+    upload_package_sizes,
+)
 
 
-def get_package_path(glob_pattern):
-    package_paths = glob.glob(glob_pattern)
-    if len(package_paths) > 1:
-        raise Exit(code=1, message=color_message(f"Too many files matching {glob_pattern}: {package_paths}", "red"))
-    elif len(package_paths) == 0:
-        raise Exit(code=1, message=color_message(f"Couldn't find any file matching {glob_pattern}", "red"))
-
-    return package_paths[0]
-
-
-def _get_deb_uncompressed_size(ctx, package):
-    # the size returned by dpkg is a number of bytes divided by 1024
-    # so we multiply it back to get the same unit as RPM or stat
-    return int(ctx.run(f'dpkg-deb --info {package} | grep Installed-Size | cut -d : -f 2 | xargs').stdout) * 1024
-
-
-def _get_rpm_uncompressed_size(ctx, package):
-    return int(ctx.run(f'rpm -qip {package} | grep Size | cut -d : -f 2 | xargs').stdout)
+@task
+def check_size(ctx, filename: str = 'package_sizes.json', dry_run: bool = False):
```

Comment:
Is this task runnable locally ?
Could it be updated to compare two arbitrary commits ? (I'm in particular interested in comparing two stable releases, or two release candidates)

Reply:
The package sizes are stored on `build-stable` so if we have access to it, it can work.
The current version expects all versions from the template to be present, and only compares what is locally from what is stored in the S3 json.
The evolution you ask for are possible through some enhancement in the task